### PR TITLE
Add CI to validate extensions.json

### PR DESCRIPTION
**Pull request by jtbandes**
### Changelog
None

### Docs

None

### Description

Adds validation script to avoid future errors in extensions.json:
- Check SHA matches
- Check that some required files are present
- Check that readme & changelog point at markdown (text) files rather than HTML or something else
- Warn if files in the .foxe are not compressed (https://github.com/foxglove/create-foxglove-extension/pull/147)

Checking the `id` values requires some more attention (and maybe changes to the app) so not solving that in this PR.

Example output:
image

### CODE REVIEW

File: validate.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  return { publisher: res[1], name: res[2]! };
+}
+
+async function main() {
+  for (const extension of registryJson) {
+    console.log(`Validating ${extension.id}...`);
+    const foxeResponse = await fetch(extension.foxe);
+    if (foxeResponse.status !== 200) {
+      throw new Error(
+        `Extension ${extension.id} has invalid foxe URL. Expected 200 response, got ${foxeResponse.status}`
```
Comment (indirectlylit) [inline validate.ts:20]: nit: maybe include `extension.foxe`  in this error?

**Comment by indirectlylit** [comment]
We could/should also enforce changelog and readme content in the form of `https://raw.githubusercontent.com/...`. Most are like this, but some use `https://github.com/` domains.

File: validate.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  }
+  const packageJson = JSON.parse(packageJsonContent);
+  if (packageJson.name == undefined || packageJson.name.length === 0) {
+    error(`${extension.id}: Invalid extension: missing name`);
+    return;
+  }
+
+  const mainPath = packageJson.main;
+  if (typeof mainPath !== "string") {
+    error(`${extension.id}: Extension package.json main is missing`);
```
Comment (bryfox) [inline validate.ts:94]: nit: I found "Extension package.json main is missing" hard to parse for some reason. Maybe 'Invalid package.json: missing "main" field'?

**Review by bryfox — APPROVED** [review]
🎉 Couple of nitpicks, but this looks great.

I worry slightly that some extensions will never get updated with compression, that we'll get "warning fatigue" when reviewing, and that we'll miss new warnings. The only idea I can think of is to enumerate extensions already in the registry, and elevate that warning to an error if the extension isn't in that set.

File: validate.ts (hunk 2 above)
Comment (jtbandes) [reply validate.ts:94]: full disclosure: this error message was copied verbatim from the app 🙊

**Comment by jtbandes** [comment]
> The only idea I can think of is to enumerate extensions already in the registry, and elevate that warning to an error if the extension isn't in that set.

Sounds good - I can also key this on the SHA to ensure we use compression for future updates to existing extensions
